File: ems-backend/src/main/java/dev/robgleason/ems/service/DepartmentService.java
package dev.robgleason.ems.service;

import dev.robgleason.ems.dto.DepartmentDto;

import java.util.List;

public interface DepartmentService {

    DepartmentDto createDepartment(DepartmentDto departmentDto);
    DepartmentDto getDepartmentById(Long departmentId);

    List<DepartmentDto> getAllDepartments();

    DepartmentDto updateDepartment(DepartmentDto department);

    void deleteDepartment(Long departmentId);
}

File: ems-frontend/src/services/DepartmentService.js
import axios from "axios";

const REST_API_BASE_URL = "http://localhost:8080/api/departments";

export const getAllDepartments = () => axios.get(REST_API_BASE_URL);

export const createDepartment = (department) =>
  axios.post(REST_API_BASE_URL, department);

export const getDepartment = (departmentId) =>
  axios.get(REST_API_BASE_URL + "/" + departmentId);

export const updateDepartment = (departmentId, department) =>
  axios.put(REST_API_BASE_URL + "/" + departmentId, department);

export const deleteDepartment = (departmentId) => {
  return axios.delete(REST_API_BASE_URL + "/" + departmentId);
};
